internal/table/table.go

```diff
@@ -55,86 +55,81 @@ func (t *Table) Render() string {
 	return strings.TrimRight(out, linebreak)
 }
 
-// equalizeWidths equalizes the width of given columns by a given percentage of the average columns width
-func equalizeWidths(colWidths []int, total int, percentage int) ([]int, int) {
-	averageWidth := total / len(colWidths)
-	equalizer := percentage * averageWidth / 100
-	total += equalizer * len(colWidths)
-	for n := range colWidths {
-		colWidths[n] += equalizer
-	}
-	return colWidths, total
-}
-
-func rescaleColumns(colWidths []int, total, targetTotal int) []int {
-	// Calculate column widths according to the total width
-	remaining := targetTotal
-	for n, w := range colWidths {
-		cw := int(math.Floor(float64(w) / float64(total) * float64(remaining)))
-		total -= w
-		remaining -= cw
-		colWidths[n] = cw
-	}
-
-	colWidths[len(colWidths)-1] += remaining // Ensure we use up all remaining space
-	return colWidths
-}
-
-func (t *Table) calculateWidth(maxTotalWidth int) ([]int, int) {
-	// Calculate required width of each column
-	minWidth := padding * 2
+func (t *Table) calculateWidth(maxTableWidth int) ([]int, int) {
+	// Calculate total width of each column, not worrying about max width just yet
+	minTableWidth := padding * 2
 	colWidths := make([]int, len(t.headers))
-	columnTotal := 0
+	colWidthsCombined := 0
 	for n, header := range t.headers {
-		// Check header sizes
-		headerSize := utf8.RuneCountInString(header)
-		if currentSize, ok := sliceutils.GetInt(colWidths, n); !ok || currentSize < headerSize {
-			colWidths[n] = headerSize // Set width according to header size
-		}
-		// Check row column sizes
+		// Start with the header size
+		colWidths[n] = utf8.RuneCountInString(header)
+
+		// Check column sizes for each row
 		for _, row := range t.rows {
-			spanWidth := columnTotal
-			rowValueSize := 0
-			if rowValue, ok := sliceutils.GetString(row.columns, n); ok {
-				rowValueSize = utf8.RuneCountInString(rowValue)
-			}
-			if colWidths[n] < rowValueSize {
-				if len(row.columns) < len(t.headers) {
-					spanWidth += rowValueSize + (padding * 2) // This is a spanned column, so its width does not apply to the individual column
-				} else {
-					colWidths[n] = rowValueSize // Set width according to column size
-				}
+			columnValue, ok := sliceutils.GetString(row.columns, n)
+			if !ok {
+				continue // column doesn't exit because the previous column spans
 			}
-			if spanWidth > minWidth {
-				minWidth = spanWidth
+			columnSize := utf8.RuneCountInString(columnValue)
+
+			// Detect spanned column info
+			rowHasSpannedColumn := len(row.columns) < len(t.headers)
+			spannedColumnIndex := len(row.columns) - 1
+
+			if rowHasSpannedColumn && n == spannedColumnIndex {
+				// Record total row size as minTableWidth
+				minTableWidth = mathutils.MaxInt(minTableWidth, colWidths[n]+columnSize+(padding*2))
```

Comment:
```suggestion
				minTableWidth = mathutils.MaxInt(minTableWidth, colWidthsCombined+columnSize+(padding*2))
```

Reply:
This would include the columns this column is spanning. But good point, it should include all preceding columns. Will fix, thanks!